recipe/meta.yaml

```diff
@@ -63,6 +69,54 @@ test:
   imports:
     - astropy
 
+outputs:
+  - name: astropy-base
+    build:
+      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+
+# build this exactly once, on the oldest supported python, to avoid migration issues
+{% if py==min_py and linux64 %}
+  - name: astropy
+    build:
+      noarch: generic
+      skip: true  # [(not py==310) and (not linux64)]
+    requirements:
+      run:
+        - {{ pin_subpackage('astropy-base', max_pin="x.x.x") }}
+        - matplotlib-base >=3.5.0,!=3.5.2
+        - certifi
+        # We also need numpy from dask[array] but we are already going to install that.
+        - dask-core >=2022.8.1
+        - h5py
+        - pyarrow >=10.0.1
+        - beautifulsoup4
+        - html5lib
+        - bleach
+        - pandas >=2.0
+        - sortedcontainers
+        - pytz
+        - jplephem
+        - mpmath
+        - bottleneck
+        # TODO: Decide if we need to add the aiohttp dependancy for fsspec[http]
```

Comment:
We need to make a decision on this. What do we use the http extra of fsspec for?

Reply:
opening fits files via http to read only sections off them.

 fsspec doesn't really have any core dependencies, so we do need to state HTTP. And IMO we should state s3fs here, too, as it's an indirect dependency.

Reply:
Why would we install s3fs and not also all the other cloud provider plugins? s3fs is only a test dependency of astropy no?

I am not adverse to adding aiohttp for fsspec's http support but it feels weird to cherry pick which services we install for.

What other ones from the extras am I missing? I thought I just have all the ones which aren't test deps and which make sense (other than aiohttp which I called out here)

Reply:
> Why would we install s3fs and not also all the other cloud provider plugins?

Because that's what we developed the feature for, and that's what we test. So please at least install that, but I'm not opposed to add others (if we do h5py, in this day and age we should do these, too).

Reply:
But my main point from below, please keep it a copy paste list from extras. If pytest or ipython don't need to be listed then clean it up from astropy first. As for `asdf-astropy`, if the circular dependency can be solved by moving it to depend on astropy-base, then definitely include it as well.

Reply:
I think this list and the [all] list in core have a subtly different function. I do not think we need the ipython, jupyter or pytest deps here. 

I will begrudgingly concede aiohttp and s3fs even though I think it's weird.

Reply:
My point of keeping the lists identical is that it will be much easier to update the required dependencies for a new version by just simply diffing the two lists rather than going into full archaeology mode and checking if something was purposefully left out here, and its usage still hasn't changed to require the inclusion.

Reply:
I see that. Perhaps we can come up with *yet another* extra on core, maybe even called "conda" to be explicit?

Reply:
Is it only ipython and pytest that you want to not include? As I said those should not cause any trouble for the recipe.
And pytest can be seen as a runtime dependency, too as the test runner is still a thing, so that is just as legitimate dependency as h5py.

Reply:
`pytest` isn't in `[all]` just [test](https://github.com/astropy/astropy/blob/main/pyproject.toml#L96), so I don't think we should include it.

I do not understand why we would include ipython or jupyter at all, but there again I don't understand why they are in `[all]` either, for that matter.

Reply:
ahh, OK, I was looking at a few weeks old version of the file. I agree then to not include it for now 

(but IMO it should be added back in, as the test runner still uses it, but that's an upstream issue now)